### PR TITLE
[Performance] Makes bot salute not do thousands of hearers_in_view calls

### PR DESCRIPTION
**Pull request by Azarak**
## About The Pull Request
It now uses view() instead, and is cooldowned per each check, instead of per each successful salute

EDIT: I've changed so the commmissioned bots invoke salutes for even more performance, as AnturK suggested. Changed the variable can_salute to can_get_salute to reflect that behaviour

## Why It's Good For The Game
Fast server go brrroom

## Changelog
:cl:
tweak: Bots have now slightly less respect for authority bots
/:cl:

### CODE REVIEW

**Comment by AnturK** [comment]
Why is this even done by every bot, just put it on these snowflake ones that start this.

**Comment by Azarak** [comment]
> 
> 
> Why is this even done by every bot, just put it on these snowflake ones that start this.

Good idea, will do

File: code/modules/mob/living/simple_animal/bot/bot.dm [hunk 1]
```diff
@@ -273,12 +273,12 @@
...
 
-	if(!commissioned && can_salute)
-		for(var/mob/living/simple_animal/bot/B in get_hearers_in_view(5, get_turf(src)))
-			if(B.commissioned)
-				visible_message("<b>[src]</b> performs an elaborate salute for [B]!")
-				can_salute = FALSE
-				addtimer(VARSET_CALLBACK(src, can_salute, TRUE), salute_delay)
+	if(commissioned && can_get_salute)
+		can_get_salute = FALSE
+		addtimer(VARSET_CALLBACK(src, can_get_salute, TRUE), salute_delay)
```
Comment (Rohesie) [inline code/modules/mob/living/simple_animal/bot/bot.dm:278]: Could you use one of the `cooldowns.dm` systems for standardization?
You can either keep the a variable such as `salute_delay` to store the cooldown, or you can just use the timer cooldown and eschew that.

Comment (Azarak) [reply code/modules/mob/living/simple_animal/bot/bot.dm]: Never used cooldowns.dm before. I think I did it right?

File: code/modules/mob/living/simple_animal/bot/bot.dm [hunk 2]
```diff
@@ -94,7 +94,7 @@
 	var/robot_arm = /obj/item/bodypart/r_arm/robot
 
 	var/commissioned = FALSE // Will other (noncommissioned) bots salute this bot?
-	var/can_salute = TRUE
+	var/next_salute_check = 0
```
Comment (Rohesie) [inline code/modules/mob/living/simple_animal/bot/bot.dm:97]:
```suggestion
	COOLDOWN_DECLARE(next_salute_check)
```

File: code/modules/mob/living/simple_animal/bot/bot.dm (hunk 1 above)
Comment (Rohesie) [reply code/modules/mob/living/simple_animal/bot/bot.dm]: This is good!

**Review by Rohesie — APPROVED** [review]

**Pull request merged**